dashboard/lib/client_state.rb

```diff
@@ -13,9 +13,8 @@ def initialize(session, cookies)
 
   # Resets all client state (level progress, lines of code, videos seen, etc.)
   def reset
-    # Also reset unencrypted cookies in case we are in a rolled-back state.
-    session[:lines] = cookies[:lines] = nil
-    session[:progress] = cookies[:progress] = nil
+    cookies[:lines] = nil
```

Comment:
This class is only intended manage state that is potentially needed on the client;  I didn't want to whack other session or cookie state that people might add for other reasons.  (Suppose, for example, someone added session state for tracking the number of times the user has visited the site-- you wouldn't want that to get whacked when client progress was reset.)